### DISCUSSION TITLE
Conventional Commits

### DISCUSSION THREAD

**Discussion by doxxx** [bagetter/BaGetter#124] [General]
May I suggest [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) to make your changelogs more readable?

**Comment by seriouz** [comment]
I had the same thought after seeing the generated change log.
The important part is to format the PR commit messages during the squash accordingly.

**Comment by Regenhardt** [comment]
Any idea if the current release action properly handles conventional commits?

**Reply by seriouz** [reply]
Have a look at the default config of git-cliff https://github.com/orhun/git-cliff/blob/main/cliff.toml#L80
To answer it shortly: Yes it should work out of the box!

**Reply by Regenhardt** [reply]
Oh, was totally looking at the wrong workflow step there. So I just push a new tag and the workflow should do its thing, right?

**Reply by seriouz** [reply]
Yes. We'll see how good the multi line merged messages will appear. Tag the latest commit in the main branch with `v1.2.0`.

**Reply by Regenhardt** [reply]
I pushed the new tag, but I can't see a workflow running or a new Release. Do I have to create the release via Github? I expected the pipeline to do that.

**Reply by seriouz** [reply]
It should run automatically when pushing a Git tag. Try to remove the v1.2.0 and re-push it.

**Reply by Regenhardt** [reply]
This one worked, changelog looks good, thanks✌.

**Comment by Regenhardt** [comment]
Looks like a squashed commit will only appear as a single change in the changelog, even if there were multiple commits with conventional commit messages squashed into it. Which kinda makes sense as that is a purely github thing to do and conventional commits aren't modelled after github. I guess we have to choose between squashing into a single change or rebase-merge if we want to keep the changes separate.

**Reply by doxxx** [reply]
Yes, that's what I would expect. One commit -> one change as far as conventional commits is concerned. If you're merging a branch into main that has multiple distinct commits/changes, it would be best to rebase it onto main and then do a fast-forward merge so that each of those distinct changes appear as distinct commits on main.
